data-prepper-plugins/parse-json-processor/src/main/java/org/opensearch/dataprepper/plugins/processor/parse/AbstractParseProcessor.java

```diff
@@ -29,6 +31,7 @@
 
 public abstract class AbstractParseProcessor extends AbstractProcessor<Record<Event>, Record<Event>> {
     private static final Logger LOG = LoggerFactory.getLogger(AbstractParseProcessor.class);
+    private static final String PROCESSING_ERRORS = "processingErrors";
```

Comment:
I think we should differentiate our metrics between two situations. The first are expected errors. These are bad input that we know will happen. The second are unexpected errors. These are probably bugs.

I think we could have a metric for the first - `parseErrors`. If we want a metric for the second, this could be named something like `processingFailures`. Maybe we'd use this in other processors as well as time goes on.

Reply:
Sure we can do that